cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/TopicController.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.controller.cluster.resources;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.EnvVarBuilder;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.strimzi.controller.cluster.ClusterController;
+import io.vertx.core.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the topic controller deployment
+ */
+public class TopicController extends AbstractCluster {
+
+    public static final String KIND = "topic";
+
+    private static final String NAME_SUFFIX = "-topic-controller";
+
+    // Topic Controller configuration
+    protected TopicControllerConfig config;
+
+    // Port configuration
+    protected static final int HEALTHCHECK_PORT = 8080;
+    protected static final String HEALTHCHECK_PORT_NAME = "hcheck-port";
+
+    // Configuration defaults
+    protected static final String DEFAULT_IMAGE = "strimzi/topic-controller:latest";
+    protected static final int DEFAULT_REPLICAS = 1;
+    protected static final int DEFAULT_HEALTHCHECK_DELAY = 10;
+    protected static final int DEFAULT_HEALTHCHECK_TIMEOUT = 5;
+    protected static final int DEFAULT_ZOOKEEPER_PORT = 2181;
+    protected static final int DEFAULT_BOOTSTRAP_SERVERS_PORT = 9092;
+    protected static final String DEFAULT_FULL_RECONCILIATION_INTERVAL = "15 minutes";
```

Comment:
Is 15 minutes really the default we want? Seems quite long. Is the topic reconciliation expensive?

Reply:
That was the default I set originally, I think. I suppose there's no reason why it can't be shorter. But we should check whether vertx does fixed interval or fixed delay in `setPeriodic()` (I suspect it's fixed interval), so we understand the consequences if the reconciliation takes longer than the interval. `Inflight` should prevent messes, but it won't prevent a pile up of threads all trying to reconcile at once.

Reply:
Good point. Vert.x docu says:
_Keep in mind that the timer will fire on a periodic basis. If your periodic treatment takes a long amount of time to proceed, your timer events could run continuously or even worse : stack up._

So we might want to use instead of `setPeriodic` just `setTimer` and set a new time at the end of each reconciliation. That said, 15 minutes still sounds like a lot of time. Especially since I assume lot of the issues we get from the topic being created for too long end up as a task for reconciliation? Or are these handled differently?